sos/plugins/pmem.py

```diff
@@ -0,0 +1,77 @@
+# This file is part of the sos project: https://github.com/sosreport/sos
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# version 2 of the GNU General Public License.
+#
+# See the LICENSE file in the source distribution for further information.
+
+from sos.plugins import Plugin, RedHatPlugin, UbuntuPlugin, DebianPlugin
+
+
+class Pmem(Plugin, RedHatPlugin, UbuntuPlugin, DebianPlugin):
+    """Persistent Memory devices
+    """
+
+    plugin_name = 'pmem'
+    profiles = ('storage', 'hardware', 'memory')
+
+    def setup(self):
+        self.add_cmd_output([
+            "ndctl --version",
+            "ndctl list -B",
+            "ndctl list -D",
+            "ndctl list -iD",
+            "ndctl list -iDF",
+            "ndctl list -R",
+            "ndctl list -iR",
+            "ndctl list -M",
+            "ndctl list -iM",
+            "ndctl list -N",
+            "ndctl list -iN",
+            "ndctl list -NX",
+            "ndctl list -iNX",
+            "ndctl list -NRD",
+            "ndctl list -iNRD",
+            "ndctl list -NuRD",
+            "ndctl list -iNuRD"
```

Comment:
The `ipmctl` commands are reasonably self-explanatory, but these are not: could we either use long options here, or include some comments that at least gives a basic sense of what is being done, and links to or references sources for further information?

When a plugin is merged in `sos`, we have to maintain it for as long as it remains in the tree, so the more information we have about what is being done the better.

It also looks like this list could be generated, at least partially, on-the-fly, avoiding some of the repetition and eye strain (you have a few repeating patterns in there, like `NuRD`, `NX`, `NRD` with `-N`, `-i` etc).

Reply:
I aggregated the options.  The output is JSON formatted so it can be post-processed if required.  This approach reduces the number of commands SOS executes and the amount of data collected.